source/common/http/http3/conn_pool.cc

```diff
@@ -82,7 +85,12 @@ allocateConnPool(Event::Dispatcher& dispatcher, Random::RandomGenerator& random_
         data.connection_ =
             Quic::createQuicNetworkConnection(h3_pool->quicInfo(), pool->dispatcher(), host_address,
                                               source_address, quic_stat_names, scope);
-        return std::make_unique<ActiveClient>(*pool, data);
+        Network::Connection& connection = *data.connection_;
+        auto client = std::make_unique<ActiveClient>(*pool, data);
```

Comment:
Would it make sense to move this to after the if () (so that we don't create the client if we don't end returning using it)?

Reply:
The connection is created during this construction.

Reply:
Ah, interesting. Thanks!